libbpf-tools/fsdist.c

```diff
@@ -0,0 +1,373 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2021 Hengqi Chen
+//
+// 20-May-2021   Hengqi Chen   Created this.
+#include <argp.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+
+#include "fsdist.h"
+#include "fsdist.skel.h"
+#include "trace_helpers.h"
+
+#define warn(...) fprintf(stderr, __VA_ARGS__)
+
+enum fs_kind {
+	BTRFS,
+	EXT4FS,
+	NFS,
+	XFS,
+};
+
+static volatile sig_atomic_t exiting;
+
+static enum fs_kind fs_kind = EXT4FS;
+static const char *fs_name = "ext4";
+static bool emit_timestamp = false;
+static bool timestamp_in_ms = false;
+static pid_t target_pid = 0;
+static int interval = 99999999;
+static int count = 99999999;
+static bool verbose = false;
+
+const char *argp_program_version = "fsdist 0.1";
+const char *argp_program_bug_address =
+	"https://github.com/iovisor/bcc/tree/master/libbpf-tools";
+const char argp_program_doc[] =
+"Summarize btrfs/ext4/nfs/xfs operation latency.\n"
+"\n"
+"Usage: fsdist [-h] [-x] [-T] [-m] [-p PID] [interval] [count]\n"
+"\n"
+"EXAMPLES:\n"
+"    fsdist                     # show ext4 operation latency as a histogram\n"
+"    fsdist -x nfs -p 1216      # trace nfs operation with PID 1216 only\n"
```

Comment:
-x seems like a weird flag for filesystem type, why not `-t`, similar to how `mount -t nfs` works?

Reply:
My first thought was to avoid `-t` since it was used for `TID` and `timestamp` in other tools.

Will change it to `-t` just like the `mount` utility.

Reply:
My first thought was to avoid `-t` since it was used for `TID` and `timestamp` in other tools.

Will change it to `-t` just like the `mount` utility.